Sage/carbon
# feat(typography): allow listitem to inherit variant from parent list

In order to reduce replication of code, ListItem now receives information on the variant of the parent List item via context; setting variant on List will pass it to child ListItem components.

Resolves #5413

### Proposed behaviour

`ListItem` now inherits `variant` directly from it's parent `List` component

### Current behaviour

Customers have to manually define the `variant` on each `ListItem`

### Checklist


#### QA


### Testing instructions

Use Storybook to view the new approach/component structure. Additionally, verify that a deprecation warning is thrown if `variant` is added to any `ListItem`
